File: Server/src/main/core/game/global/action/LadderAchievementCheck.java
package core.game.global.action;

import core.game.node.entity.player.Player;

/**
 * The interface Ladder achievement check.
 */
interface LadderAchievementCheck {
    /**
     * Check achievement.
     *
     * @param player the player
     */
    default void checkAchievement(Player player) {
        return;
    }
}

File: Server/src/main/core/game/global/action/LadderAchievementCheck.kt
package core.game.global.action

import core.game.node.entity.player.Player

/**
 * Ladder achievement check
 *
 * @constructor Ladder achievement check
 */
internal interface LadderAchievementCheck {
    /**
     * Check achievement
     *
     * @param player
     */
    fun checkAchievement(player: Player) {
        return
    }
}
